config/puma.rb

```diff
@@ -16,6 +16,9 @@
 #
 environment ENV.fetch("RAILS_ENV") { "development" }
 
+# Specifies the `pidfile` that Puma will use.
+pidfile ENV.fetch("PIDFILE") { "tmp/pids/server.pid" }
```

Comment:
I found this from [Rails diff](railsdiff.com) which is a pretty cool website for checking config changes between rails versions.